### ISSUE TITLE
Fix incorrect line numbers in selector-no-qualifying-type

### ISSUE THREAD

**Issue by glen-84** [stylelint/stylelint#4410]
> Clearly describe the bug

The reported error locations (line, column) are often incorrect in Sass (scss) files.

> Which rule, if any, is the bug related to?

`selector-no-qualifying-type`

> What CSS is needed to reproduce the bug?

```scss
@import "../../../settings";

// Foundation

@import "~foundation-sites/scss/foundation";

// Mixins

@import "../../../mixins/fonts";
@import "../../../mixins/module";

.tournament-matches {
    @include module(
        $bg: $background-light,
        $border: $background,
        $divider: $background-dark
    );

    .global-results {
        flex-wrap: nowrap;
        display: flex;
        justify-content: flex-end;
        flex-direction: row-reverse;
        margin-left: 20px;

        .switch {
            margin-left: 5px;
        }
    }

    .match-list {
        h3 {
            @include font-heading;
            color: $text-secondary;
        }

        .match-date {
            @include font-regular($size-big);
            color: $text-discreet;
            padding: 20px 0 10px;
        }

        .match {
            background-color: $background-lighter;
            margin-bottom: 1px;

            .match-score {
                @include font-heading;
                text-align: center;
                color: $text-discreet;
                position: relative;
                display: flex;
                align-items: center;
                justify-content: center;

                .loss {
                    color: $text-discreet;
                }

                .show-result {
                    @include font-small($size-regular);
                    opacity: 0;
                    position: absolute;
                    left: 0;
                    width: 100%;
                    display: flex;
                    justify-content: center;
                    align-items: center;
                    top: 50%;
                    padding: 10px;
                    margin-top: -15px;
                    cursor: pointer;
                }

                .button {
                    height: 30px;
                    padding: 0;
                }
            }

            &.result-hidden {
                .match-score span {
                    opacity: 0;
                    visibility: hidden;
                }

                .show-result {
                    opacity: 1;
                    visibility: visible;
                }
            }

            .match-info {
                padding: 10px;
                white-space: nowrap;

                .match-tournament {
                    color: $text-secondary;
                    overflow: hidden;
                    text-overflow: ellipsis;
                }

                .match-connect {
                    margin: 0 5px;
                    color: $text-secondary;
                }
            }

            .match-status {
                display: flex;
                align-items: center;
                align-self: stretch;
                justify-content: center;
                color: $text-light;
                white-space: nowrap;

                time.relative {
                    @include font-small;
                    margin-left: 20px;
                    margin-right: 10px;
                    width: 60px;
                    color: $text-secondary;
                }
            }

            &.live .match-status span {
                @include font-small;
                background-color: $gosu-red;
                padding: 3px 5px;
            }

            .match-video {
                display: flex;
                text-align: center;
                justify-content: center;
                align-items: center;
            }

            .has-video::before {
                font-family: $font-icons;
                content: "\f16a";
                color: $text-discreet;
                font-size: $size-large;
            }

            .match-odds {
                display: flex;
                text-align: center;
                justify-content: center;
                align-items: center;
            }

            .has-odds {
                @include font-regular($size-smaller);
                color: $text-discreet;
                border: 2px solid $text-discreet;
                padding: 0 2px;
                font-weight: bold;
                cursor: default;
            }

            .match-gosubet {
                display: flex;
                text-align: center;
                justify-content: center;
                align-items: center;

                span {
                    background-color: $background;
                    border-radius: 50%;
                    width: 20px;
                    height: 20px;
                    cursor: default;
                    display: inline-block;
                    text-align: center;
                    line-height: 20px;
                    color: $text-discreet;
                    margin: 0 2px;

                    &.winner {
                        background-color: $gosu-green;
                        color: $background;
                    }

                    &.bet {
                        border: 1px solid $gosu-green;
                        color: $gosu-green;
                        line-height: 18px;
                    }
                }
            }

            &.multi-opponent {
                .match-score span {
                    @include font-small;
                    line-height: normal;
                }
            }
        }

        .multi-opponent {
            .match-link {
                flex-direction: column;
            }

            .match-date {
                margin-bottom: 0;
            }

            .time {
                margin-top: 10px;
            }

            .reveal {
                margin-top: -2px;
            }

            .score {
                @include font-regular($size-regular);
                width: 100%;
                color: $gosu-green;
                margin-top: 10px;
                margin-bottom: 6px;

                img {
                    @include avatar-regular(25);
                    border-radius: 50%;
                    border: 2px solid $gosu-green;
                    margin-right: 5px;
                }
            }

            .result {
                min-height: 25px;
                line-height: 25px;
                overflow-x: visible;
            }
        }

        .illustrated-placeholder {
            margin: 0;
        }
    }
}

.matches-subheader {
    label {
        display: flex;
        justify-content: center;
        align-items: center;
        margin-top: 10px;

        span:not(.form-error) {
            display: flex;
            white-space: nowrap;
            align-items: center;
            margin: 0 10px 0 0;
        }
    }
}
```

OR

```scss
div {
    div {
        div.d {
            color: red;
        }
    }
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-no-qualifying-type": true
  }
}
```

> Which version of stylelint are you using?

`11.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "assets/styles/**/*.scss"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS syntax.

> What did you expect to happen?

Correct error locations (line, column).

> What actually happened (e.g. what warnings or errors did you get)?

Incorrect error locations (line, column).

![image](https://user-images.githubusercontent.com/261509/68750551-b9818680-0608-11ea-8876-1880a62a4ec1.png)

(just one example)

**Comment by hudochenkov** [comment]
Behaviour is inconsistent.

If it's three levels, then it report wrong:

https://stylelint.io/demo#N4Igxg9gJgpiBcIoEsBuACYAdAdu-6KG2eBZRAdFJrmXQZADYQBO86LMUA3LfegF8++ITgEgANCADOATxwAXAIYAPBDLDTpk8BBwAzZAHN1JfFhAsAroxjSL7MwQvSYtsAtYBaHBC8BHKyVGZH1ZZBwjLwVZAAcYB3QFawTSUXEBIA

If it's two levels, then it reports correctly:

https://stylelint.io/demo#N4Igxg9gJgpiBcIoEsBuACYAdAdu-6KqAdFJrgZepADYQBO869MUA3BQQL65cgA0IAM4BPHABcAhgA8EwsEKEDwEHADNkAcznY86LCHoBXGjCEGmuygaExTYcQwC0OCE4CORyTWRqRyHE0ncREABxgLdHFjCL0eHD4uIA
